strax/storage/s3.py

```diff
@@ -162,56 +165,65 @@ class S3Saver(strax.Saver):
     def __init__(self, key, metadata, meta_only,
                  **kwargs):
         super().__init__(metadata, meta_only)
-        self.s3client = boto3.client(**kwargs,
-                                     service_name='s3')
+        self.s3 = boto3.client(**kwargs,
+                               service_name='s3')
 
         # Unique key specifying processing of a run
-        self.key = key
-
-        self.s3client.create_bucket(Bucket=self.key)  # Does nothing if exists
+        self.strax_unique_key = key
 
     def _save_chunk(self, data, chunk_info):
-        filename = '%06d' % chunk_info['chunk_i']
+        # Keyname
+        key_name = f"{self.strax_unique_key}/{chunk_info['chunk_i']:06d}"
 
+        # Save chunk via temporary file
         with tempfile.SpooledTemporaryFile() as f:
             filesize = strax.save_file(f,
                                        data=data,
                                        compressor=self.md['compressor'])
             f.seek(0)
-            self.s3client.upload_fileobj(f, self.key, filename)
+            self.s3.upload_fileobj(f,
+                                   BUCKET_NAME,
+                                   key_name)
 
-        return dict(filename=filename, filesize=filesize)
+        return dict(key_name=key_name,
```

Comment:
OK, this is where key_name gets set. Got it. So it's like filename except that it currently has the strax key attached in front of it.